### ISSUE TITLE
issue with manual_rotation2 for LiCoO2

### ISSUE THREAD

**Issue by WalterWhite1611** [Chengcheng-Xiao/RotSph#4]
I am trying to align octahedrals of LiCoO2 along global cartesian axes with `construct_euler.py` script by running `!python3 construct_euler.py -0.68715 -0.20655 -1.79943 -1.50865 1.03646 0.63448  T`, Euler angles are `[ 146.9685887   -14.44426056 -163.57609429]`. Then with `atomsk` I rotated the POSCAR with the Euler angles in z,y,x order. But I am not getting aligned structure.
[POSCAR.txt](https://github.com/user-attachments/files/18141870/POSCAR.txt)

**Comment by Chengcheng-Xiao** [comment]
If I remember correctly, there are multiple definitions of Euler angles, and RotSph only works with one of them. If you want to use other codes and rotate POSCAR, then you need to make sure the same definition is used.

**Comment by WalterWhite1611** [comment]
> If I remember correctly, there are multiple definitions of Euler angles, and RotSph only works with one of them. If you want to use other codes and rotate POSCAR, then you need to make sure the same definition is used.

yes I used `ZYX` Euler angles, which is implemented in `construct_euler.py` script of RotSph. You kindly verify this and say how to solve this for any kind of structure? I used this script for CrI3 supercell, and then it's working perfectly.

**Comment by Chengcheng-Xiao** [comment]
Can you explain how `-0.68715 -0.20655 -1.79943 -1.50865 1.03646 0.63448` were obtained?

**Comment by WalterWhite1611** [comment]
> Can you explain how `-0.68715 -0.20655 -1.79943 -1.50865 1.03646 0.63448` were obtained?

![image](https://github.com/user-attachments/assets/0177628c-8fb6-46a4-8429-3154c0f8d667)
selecting Co[1] - O[2] for both local c and a axis with different O[2] coordinate. Kindly verify this, is it correct or not by opening on VESTA. I extracted coordinates from VESTA

**Comment by Chengcheng-Xiao** [comment]
So, your Co[1] sits at the origin ([0,0,0] in Cartesian coordinates) and if I add two atoms located at (-0.68715 -0.20655 -1.79943) and (-1.50865 1.03646 0.63448), this is the structure I get:

<img width="442" alt="image" src="https://github.com/user-attachments/assets/9ddfcf80-ee8e-46ce-a7e8-899cb3fcf025" />

As you can see the bonds they form to Co1 do not align with any Co-O bonds and because of this I don't think you have the correct coordinates.

**Comment by WalterWhite1611** [comment]
yes I get your point. But I took consider different cartesian coordinate of Co[1], see the image 
![image](https://github.com/user-attachments/assets/16183ec0-0823-43dc-9970-93f794bbed68)
with [0,0,0] cartesian coordinate , see O[2] (x,y,z) coordinate 
![image](https://github.com/user-attachments/assets/51268710-dcb0-458f-8ace-91f97a2742a6)
but still I am unable to align octahedra. Can you kindly say from the POSCAR what should be the local c and a axis coordinate?

**Comment by WalterWhite1611** [comment]
> So, your Co[1] sits at the origin ([0,0,0] in Cartesian coordinates) and if I add two atoms located at (-0.68715 -0.20655 -1.79943) and (-1.50865 1.03646 0.63448), this is the structure I get:
> 
> <img alt="image" width="442" src="https://private-user-images.githubusercontent.com/24631005/396103301-9ddfcf80-ee8e-46ce-a7e8-899cb3fcf025.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ5NDI4ODIsIm5iZiI6MTczNDk0MjU4MiwicGF0aCI6Ii8yNDYzMTAwNS8zOTYxMDMzMDEtOWRkZmNmODAtZWU4ZS00NmNlLWE3ZTgtODk5Y2IzZmNmMDI1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIzVDA4Mjk0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThiMmVhNjYxYTEwN2YxZDIyM2JkODRlOGYzMTI2NjBkNDRmMTAyMWJjNmRkMmIyZmNmYzFjNmM3YmMwZDAyZTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.2TCB0HmPztvr4IT_TCDkpsFMuoq3vsFazBrykckrUdg">
> As you can see the bonds they form to Co1 do not align with any Co-O bonds and because of this I don't think you have the correct coordinates.

Can you kindly say what should be the exact coordinates to align this octahedra?

**Comment by Chengcheng-Xiao** [comment]
Cannot reproduce your results:
<img width="537" alt="image" src="https://github.com/user-attachments/assets/340c6055-df4f-4975-9dd4-b99d13e8a650" />
Please make sure you are using the correct structure.

**Comment by WalterWhite1611** [comment]
[POSCAR.txt](https://github.com/user-attachments/files/18251161/POSCAR.txt)
I dont know where is the issue is occuring. Here I have attached POSCAR file ,downloaded from Materials Project, still culd not able to align. Kindly you verify where is the issue?

**Comment by Chengcheng-Xiao** [comment]
Sorry I cannot determine the problem as I cannot reproduce your result (i.e., the directions I found are different from yours using the same structure).

However, I do believe this is a simple case of finding the right directions.
Please double check and make sure you are using the correct structure with the correct directions.

**Comment by WalterWhite1611** [comment]
> Sorry I cannot determine the problem as I cannot reproduce your result (i.e., the directions I found are different from yours using the same structure).
> 
> However, I do believe this is a simple case of finding the right directions. Please double check and make sure you are using the correct structure with the correct directions.

The structure is correct, I downloaded from MP. What do you mean by choosing right directions? you mean choosing specific bond to align along z axis or x axis!!

**Comment by Chengcheng-Xiao** [comment]
With your structure:
```
LiCoO2
1.0
	1.4268725587853039 0.8238045651055055 4.6748962972028387
	-1.4268735849996073 0.8238048146972278 4.6748966644229135
	-0.0000003165135772 -1.6476119853483864 4.6748967296837227
  Li Co O
  1 1 2
 Direct
   0.5000000960575463 0.5000000935488410 0.5000000956882831
   0.0000001050370462 0.0000001066686071 0.0000001045246551
   0.2606809464359688 0.2606811021219804 0.2606811619358870
   0.7393188524694385 0.7393186976605717 0.7393186378511745
```
Try:
```
python construct_euler.py 0.00000  -1.64761   1.01892 1.42687   0.82381   1.01892 T
```
And rotate the structure according to the results of the script.

This is the best I can help as your use of the code is not strictly related to RotSph and I cannot reproduce your results.
Good luck.🍀

**Comment by WalterWhite1611** [comment]
Thank you Chengcheng. I solved that issue. My old version vesta was showing wrong coordinates. I have updated.
